src/main/java/net/snowflake/ingest/streaming/internal/ChannelCache.java

```diff
@@ -44,13 +67,73 @@ void addChannel(SnowflakeStreamingIngestChannelInternal<T> channel) {
   }
 
   /**
-   * Returns an iterator over the (table, channels) in this map.
+   * Get the last flush time for a table
+   *
+   * @param fullyQualifiedTableName fully qualified table name
+   * @return last flush time in milliseconds
+   */
+  Long getLastFlushTime(String fullyQualifiedTableName) {
+    FlushInfo tableFlushInfo = this.tableFlushInfo.get(fullyQualifiedTableName);
+    if (tableFlushInfo == null) {
+      throw new SFException(
+          ErrorCode.INTERNAL_ERROR,
+          String.format("Last flush time for table %s not found", fullyQualifiedTableName));
+    }
+    return tableFlushInfo.lastFlushTime;
+  }
+
+  /**
+   * Set the last flush time for a table as the current time
+   *
+   * @param fullyQualifiedTableName fully qualified table name
+   * @param lastFlushTime last flush time in milliseconds
+   */
+  void setLastFlushTime(String fullyQualifiedTableName, Long lastFlushTime) {
+    this.tableFlushInfo.compute(
+        fullyQualifiedTableName,
+        (k, v) -> new FlushInfo(lastFlushTime + getRandomFlushJitter(), v != null && v.needFlush));
+  }
+
+  /**
+   * Get need flush flag for a table
+   *
+   * @param fullyQualifiedTableName fully qualified table name
+   * @return need flush flag
+   */
+  Boolean getNeedFlush(String fullyQualifiedTableName) {
+    FlushInfo tableFlushInfo = this.tableFlushInfo.get(fullyQualifiedTableName);
+    if (tableFlushInfo == null) {
+      throw new SFException(
+          ErrorCode.INTERNAL_ERROR,
+          String.format("Need flush flag for table %s not found", fullyQualifiedTableName));
+    }
+    return tableFlushInfo.needFlush;
+  }
+
+  /**
+   * Set need flush flag for a table
    *
-   * @return
+   * @param fullyQualifiedTableName fully qualified table name
+   * @param needFlush need flush flag
    */
-  Iterator<Map.Entry<String, ConcurrentHashMap<String, SnowflakeStreamingIngestChannelInternal<T>>>>
-      iterator() {
-    return this.cache.entrySet().iterator();
+  void setNeedFlush(String fullyQualifiedTableName, Boolean needFlush) {
```

Comment:
why needFlush is Boolean instead of boolean?

Reply:
Both works, change to boolean.